src/main/java/org/springframework/data/mapping/PropertyPath.java

```diff
@@ -252,6 +281,11 @@ public static PropertyPath from(String source, TypeInformation<?> type) {
 			}
 		}
 
+		if (result == null) {
+			throw new IllegalStateException(
```

Comment:
Eclipse doesn't recognize that as it basically requires Spring specific knowledge in the IDE which I don't think we can/should rely on.